### DISCUSSION TITLE
Create JSX API style with react-table

### DISCUSSION THREAD

**Discussion by yeion7** [TanStack/table#2231] [General] [answered]
I would like to migrate to use `react-table` in my company because we have a lot of tables, but the current implementation use a JSX style API, I couldn't find a example about use `react-table` as primitive to expose a JSX API

Example

```
<Table data={data}>
  <ColumnGroup title="Name">
    <Column title="First Name" dataIndex="firstName" key="firstName" />
    <Column title="Last Name" dataIndex="lastName" key="lastName" />
  </ColumnGroup>
  <Column title="Age" dataIndex="age" key="age" />
  <Column title="Address" dataIndex="address" key="address" />
  <Column
    title="Action"
    key="action"
    render={(text, record) => (
      <Button>
        Delete
      </Button>
    )}
  />
</Table>
```

I don't know to much about react-table, but I'm wondering if is this possible?

**Comment by GuptaSiddhant** [comment, marked as answer]
React Table v7 is an headless API. It cannot dictate such structure. The closest thing to your requirement is [React-Table v6](https://github.com/tannerlinsley/react-table/tree/v6), which does expose `Table` JSX API but not any children JSX. You'd still need to pass information as objects.
